[0.20.0] Error: Cannot find module '/Users/xxx/test-vitest/node_modules/.pnpm/vitest@0.20.0_jsdom@20.0.0/node_modules/vitest/suppress-warnings.cjs'

### Describe the bug

In v0.20.0, run `vitest`, then: 
Error: Cannot find module '/Users/xxx/test-vitest/node_modules/.pnpm/vitest@0.20.0_jsdom@20.0.0/node_modules/vitest/suppress-warnings.cjs'

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-r2ajdl?file=package.json,vite.config.ts&initialPath=__vitest__

### System Info

```shell
System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Memory: 39.45 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 8.11.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 103.0.5060.134
    Safari: 13.1.1
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vitest/src/runtime/loader.ts
+++ b/packages/vitest/src/runtime/loader.ts
@@ -20,6 +20,7 @@ export const resolve: Resolver = async (url, context, next) => {
     if (resolved) {
       return {
         url: pathToFileURL(resolved.id).toString(),
+        shortCircuit: true,
       }
     }
   }

